tests/test_time_series_graph.py

```diff
@@ -811,6 +811,59 @@ def test_to_numpy_by_lag(self):
         self.assertEqual(adj_matrices[-1].shape, (3, 3))
         self.assertEqual(adj_matrices[-2].shape, (3, 3))
 
+    def test_get_nodes_at_lag(self):
+        self.assertEqual(['X1', 'X2', 'X3'], [node.identifier for node in self.tsdag.get_nodes_at_lag(0)])
+        self.assertEqual(
+            ['X1 lag(n=1)', 'X2 lag(n=1)', 'X3 lag(n=1)'], [node.identifier for node in self.tsdag.get_nodes_at_lag(-1)]
+        )
+
+        # same with tsdag_1
+        self.assertEqual(['X1', 'X2', 'X3'], [node.identifier for node in self.tsdag_1.get_nodes_at_lag(0)])
+        self.assertEqual(
+            ['X1 lag(n=1)', 'X2 lag(n=1)', 'X3 lag(n=1)'],
+            [node.identifier for node in self.tsdag_1.get_nodes_at_lag(-1)],
+        )
+        self.assertEqual(
+            ['X1 lag(n=2)', 'X2 lag(n=2)', 'X3 lag(n=2)'],
+            [node.identifier for node in self.tsdag_1.get_nodes_at_lag(-2)],
+        )
+
+    def test_contemporaneous_nodes(self):
+        self.assertEqual([n.identifier for n in self.tsdag.get_contemporaneous_nodes('X1')], ['X2', 'X3'])
+        self.assertEqual([n.identifier for n in self.tsdag.get_contemporaneous_nodes('X2')], ['X1', 'X3'])
+        self.assertEqual([n.identifier for n in self.tsdag.get_contemporaneous_nodes('X3')], ['X1', 'X2'])
+
+    def test_from_gml(self):
+        reconstruction = TimeSeriesCausalGraph.from_gml_string(self.tsdag.to_gml_string())
+        self.assertEqual(self.tsdag, reconstruction)
+
+        # same with deep equality
+        self.assertTrue(self.tsdag.__eq__(reconstruction, True))
+
+    def test_from_networkx(self):
+        reconstruction = TimeSeriesCausalGraph.from_networkx(self.tsdag.to_networkx())
+        self.assertEqual(self.tsdag, reconstruction)
+
+        # same with deep equality
+        self.assertTrue(self.tsdag.__eq__(reconstruction, True))
+
+    def test_from_skeleton(self):
+        reconstruction = CausalGraph.from_skeleton(self.tsdag.skeleton)
+
+        # The reconstruction won't have directions so should just match skeleton.
+
+        # Check that their networkx representations are the same.
+        self.assertTrue(networkx.utils.graphs_equal(self.tsdag.skeleton.to_networkx(), reconstruction.to_networkx()))
+        # Can check networkx as the reconstruction has no directions so networkx.Graph of CausalGraph and
+        # Skeleton will be the same.
+        self.assertTrue(
+            networkx.utils.graphs_equal(reconstruction.to_networkx(), reconstruction.skeleton.to_networkx())
+        )
+
+        # Also confirm that equality method works. Again reconstruction won't have directions so just check skeletons.
+        self.assertEqual(self.tsdag.skeleton, reconstruction.skeleton)
+        self.assertTrue(self.tsdag.skeleton.__eq__(reconstruction.skeleton, False))
```

Comment:
This is same as just calling `==`. Think you mean `self.assertFalse(self.tsdag.skeleton.__eq__(reconstruction.skeleton, True))`